uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/manager-leave/components/manager-leave-requests_test.js

```diff
@@ -1,30 +1,899 @@
+        });
+
+        it('sets the leaveStatus', function () {
+          expect(controller.filters.leaveRequest.leaveStatus).toEqual(mockStatus);
+        });
+      });
+
+      describe('getUserNameByID', function () {
+
+        var returnValue;
+
+        beforeEach(function () {
+          returnValue = controller.getUserNameByID(controller.filteredUsers[0].id);
+        });
+
+        it('returns name of the user', function () {
+          expect(returnValue).toBe(controller.filteredUsers[0].display_name);
+        });
+      });
+
+      describe('refresh::', function () {
+
+        describe('page no::', function () {
+
+          describe('when no page no is sent', function () {
+
+            beforeEach(function () {
+              controller.refresh();
+            });
+
+            it('page no is set to 1', function () {
+              expect(controller.pagination.page).toBe(1);
+            });
+          });
+
+          describe('when page no is sent', function () {
+
+            var pageNo = 5;
+            beforeEach(function () {
+              controller.refresh(pageNo);
+            });
+
+            it('page no is set to 1', function () {
+              expect(controller.pagination.page).toBe(pageNo);
+            });
+          });
+        });
+
+        describe('content loading', function () {
+
+          beforeEach(function () {
+            controller.refresh();
+          });
+
+          it('is shown', function () {
+            expect(controller.loading.content).toBe(true);
+          });
+        });
+
+        describe('contactFilters::', function () {
+
+          describe('region::', function () {
+
+            describe('when region filter has value', function () {
+
+              var mockRegion = {
+                value: 'mockvalue'
+              };
+
+              beforeEach(function () {
+                controller.filters.contact.region = mockRegion;
+                controller.refresh();
+              });
+
+              it('ContactAPI gets called with region value', function () {
+                expect(Contact.all).toHaveBeenCalledWith(jasmine.objectContaining({
+                  region: mockRegion.value
+                }), jasmine.any(Object));
+              });
+            });
+
+            describe('when region filter has no value', function () {
+
+              beforeEach(function () {
+                controller.filters.contact.region = null;
+                controller.refresh();
+              });
+
+              it('ContactAPI gets called with null', function () {
+                expect(Contact.all).toHaveBeenCalledWith(jasmine.objectContaining({
+                  region: null
+                }), jasmine.any(Object));
+              });
+            });
+          });
+
+          describe('department::', function () {
+
+            describe('when department filter has value', function () {
+
+              var mockDepartment = {
```

Comment:
I think we can skip this, we already had a discussion on this. It really makes hard to read the code at least for me.